### ISSUE TITLE
[discussion] what exactly is the point of the "processor" tab?

### ISSUE THREAD

**Issue by abdullahezzat1** [wingman-jr-addon/wingman_jr#145]
I don't really see the point of the processor tab!!

**Comment by wingman-jr-addon** [comment]
Good question!

The initial background is found on #75 with the explanation on [a blog post](https://wingman-jr.blogspot.com/2020/12/firefox-83-fix-in-progress.html). The short version is that I used to have the processing down in the "background" context of the addon. However, a change in Firefox caused the "background" context to not interact with the GPU correctly for my machine and several others. I lost several users because performance became completely unusable after Firefox updated. So what could I do? Well, I could launch a "content page" and then tunnel all of the data from the background context to the content page for actual processing. I basically had to rewrite 50% of the plugin or more and implement a client/server architecture! Additionally, I also have to add permission to have a hidden tab, which seems suspicious. I don't like it. It was necessary. But I still haven't recovered some of the users from that time. :(

The one upside to it is that it allows me to spin up more than one processing tab if I want. I tried implementing two tabs but so far one tab is more efficient than two, so I've kept it at that for now.

Does that answer your question @abdullahezzat1 ?
